drivers: serial: uart_xlnx_ps: switch over to DEVICE_MMIO API

ibirnbaum opened the pull request:
Switch the Xilinx PS UART device driver over to the use of the DEVICE_MMIO_... macros instead of just using the physical base
address from the device tree.

Signed-off-by: Immo Birnbaum <Immo.Birnbaum@weidmueller.com>

ibirnbaum committed:
drivers: serial: uart_xlnx_ps: switch over to DEVICE_MMIO API

Switch the Xilinx PS UART device driver over to the use of the
DEVICE_MMIO_... macros instead of just using the physical base
address from the device tree.

Signed-off-by: Immo Birnbaum <Immo.Birnbaum@weidmueller.com>

henrikbrixandersen reviewed (changes requested):
Looks good, thanks. Should the corresponding `MMU_REGION_FLAT_ENTRY()` calls be removed from `soc.c` for the Xilinx Zynq7000?

ibirnbaum committed:
soc: xilinx_zynq7000: remove static MMU entries for PS UARTs

Remove the static MMU entries for each available Xilinx PS UART
instance. Those are no longer required now that the PS UART
driver uses the DEVICE_MMIO API for register space mapping.

Signed-off-by: Immo Birnbaum <immo.birnbaum@weidmueller.com>

ibirnbaum commented:
> Looks good, thanks. Should the corresponding `MMU_REGION_FLAT_ENTRY()` calls be removed from `soc.c` for the Xilinx Zynq7000?

That was the actual point of the whole driver modification, and I missed it...
Removed the static MMU entries from the xc7zxxx(s)/soc.c files accordingly.
